src/lib/functions/netlify-function.js

```diff
@@ -0,0 +1,104 @@
+const BACKGROUND_SUFFIX = '-background'
+
+class NetlifyFunction {
```

Comment:
This is an entity that represents a function. It holds information about the function's runtime and its files, and contains a method for invoking the function.

The function registry will hold instances of this class.